.build/run-ci.sh

```diff
@@ -0,0 +1,186 @@
+
+nohup kubectl port-forward svc/jenkins-operator-http-example 8080:8080 &
+echo "port-forwarding running in background"
+# echo "To forward the Jenkins service to another terminal, open a new terminal window and run the following command:"
+# echo "kubectl port-forward -n ${KUBE_NS} $jenkins_pod 8080:8080"
+
+TOKEN=$(kubectl  get secret jenkins-operator-credentials-example -o jsonpath="{.data.token}" | base64 --decode)
+
+# Trigger a new build and capture the response headers
+response_headers=$(curl -i -X POST http://localhost:8080/job/k8s-e2e/buildWithParameters -u jenkins-operator:$TOKEN --data-urlencode "TEST_STAGES_TO_RUN=$INCLUDE_TEST_STAGE" 2>&1)
+
+
+queue_url=$(echo "$response_headers" | grep -i "Location" | awk -F ": " '{print $2}' | tr -d '\r')
+queue_item_number=$(basename "$queue_url")
+
+# Construct the complete URL to retrieve build information
+queue_json_url="http://localhost:8080/queue/item/$queue_item_number/api/json"
+
+# Wait for the build number to become available (querying the API)
+build_number=""
+while [ -z "$build_number" ] || [ "$build_number" == "null" ]; do
+    build_info=$(curl -s "$queue_json_url" -u jenkins-operator:$TOKEN)
+    build_number=$(echo "$build_info" | jq -r '.executable.number')
+    if [ -z "$build_number" ] || [ "$build_number" == "null" ]; then
+        echo "Build number not available yet. Waiting..."
+        sleep 5  # Adjust the polling interval as needed
+    fi
+done
+
+echo "build_number $build_number"
+BUILD_DIR=/var/lib/jenkins/jobs/k8s-e2e/builds
+POD_NAME=jenkins-example
+LATEST_BUILD=$(kubectl exec -it $POD_NAME -- /bin/bash -c "ls -t $BUILD_DIR | head -n 1" | tr -d '\r')
+
+CONSOLE_LOG_FILE="$BUILD_DIR/$build_number/log"
+
+# Define a function to check if "FINISHED" is in the consoleLog
+check_finished() {
+    kubectl exec -n $KUBE_NS $POD_NAME -- cat "$CONSOLE_LOG_FILE" | grep -q "Finished"
+}
+
+# Continuously check for "FINISHED"
+while true; do
+    if check_finished; then
+        echo "Build has finished."
+        break
+    else
+        echo "Build is still in progress."
+    fi
+    sleep 5  # Adjust the sleep interval as needed
+done
+
+LOCAL_DIR=.
+mkdir -p $LOCAL_DIR/$build_number
+kubectl cp -n $KUBE_NS $POD_NAME:$BUILD_DIR/$build_number/log $LOCAL_DIR/$build_number/log
+zip -r $LOCAL_DIR/$build_number/files.zip $LOCAL_DIR/$build_number/log
+rm $LOCAL_DIR/$build_number/log
+
+if kubectl exec -n $KUBE_NS $POD_NAME -- test -e $BUILD_DIR/$build_number/junitResult.xml; then
+    # Copy and compress the junitResult.xml file
+    kubectl cp -n $KUBE_NS $POD_NAME:$BUILD_DIR/$build_number/junitResult.xml $LOCAL_DIR/$build_number/junitResult.xml
+
+    # Create a zip archive of the copied junitResult.xml file
+    zip -j $LOCAL_DIR/$build_number/junitResult.zip $LOCAL_DIR/$build_number/junitResult.xml
+
+    # Clean up: remove the individual junitResult.xml file
+    rm $LOCAL_DIR/$build_number/junitResult.xml
```

Comment:
let's simplify this (i can do the compression later - i had something different in mind)
```suggestion
RESULTS_DIR="${CASSANDRA_DIR}/build/ci_${build_number}"
mkdir -p ${RESULTS_DIR}
kubectl cp -n $KUBE_NS $POD_NAME:${CONSOLE_LOG_FILE} ${RESULTS_DIR}/log

if kubectl exec -n $KUBE_NS $POD_NAME -- test -e $BUILD_DIR/$build_number/junitResult.xml; then
    kubectl cp -n $KUBE_NS $POD_NAME:$BUILD_DIR/$build_number/junitResult.xml ${RESULTS_DIR}/junitResult.xml
```